### PR TITLE
Add plug-in example to transform topics with XSLT after preprocess

### PR DESCRIPTION
**Pull request by jelovirt**
## Description
Add plug-in example to transform topics with XSLT after preprocess.

[Preview](https://preview-pr-486--dita-ot.netlify.app/dev/topics/plugin-preprocess-xslt.html)

## Motivation and Context
Current plug-in examples don't cover this use case.

## Type of Changes
- New feature _(non-breaking change which adds functionality)_

### CODE REVIEW

File: topics/plugin-preprocess-xslt.dita [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE task PUBLIC "-//OASIS//DTD DITA Task//EN" "task.dtd">
+<!--  This file is part of the DITA Open Toolkit project. See the accompanying LICENSE file for applicable license.  -->
+<task id="plugin-xsltparams">
+  <title>Processing topics with XSLT in preprocess</title>
+  <shortdesc>You can pass parameters from the Ant build to existing XSLT steps in both the pre-processing pipeline and
```
Comment (jelovirt) [inline topics/plugin-preprocess-xslt.dita:6]: Treat all the content as something that a Person Who Knows What They're Doing needs to go over and copy edit.

File: topics/plugin-preprocess-xslt.dita [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    certain DITA-OT transformations. This can be useful if you want to make the parameters available as global
+      <xmlelement>xsl:param</xmlelement> values within XSLT overrides.</shortdesc>
+  <taskbody>
+    <steps>
+      <step>
+        <cmd>Create a plug-in descriptor file <filepath>plugin.xml</filepath> that imports a new Ant buildfile
```
Comment (jelovirt) [inline topics/plugin-preprocess-xslt.dita:12]: Do we have a topic somewhere that describes how to write a plug-in? Basically there are two ways:

1. Create a directory somewhere and that's your plug-in source directory/repo.
2. Create a plug-in directory under `$DITA_HOME/plugins`.

The latter is a quick and dirty but simple way of doing it. Then there's the hacker version

3. Create a directory somewhere and that's your plug-in source directory/repo and then use `ln` command to create a symbolic link to plug-in directory under `$DITA_HOME/plugins`.

File: topics/plugin-common.dita [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE task PUBLIC "-//OASIS//DTD DITA Task//EN" "task.dtd">
+<!--  This file is part of the DITA Open Toolkit project. See the accompanying LICENSE file for applicable license.  -->
+<task id="plugin-common">
+  <title>Common plug-in example components</title>
+  <taskbody>
+    <steps>
+      <step id="install-plugin">
+        <cmd>Install the plug-in.</cmd>
```
Comment (jelovirt) [inline topics/plugin-common.dita:9]: This naturally depends on how you've been developing the plug-in. If you just created a custom plug-in directory under `$DITA_HOME/plugins`, you can run `dita install`. If you have your plug-in code somewhere else, you need to ZIP it and then run `dita install $ZIP_PATH`.

File: topics/plugin-preprocess-xslt.dita (hunk 2 above)
Comment (infotexture) [reply topics/plugin-preprocess-xslt.dita:12]: > Do we have a topic somewhere that describes how to write a plug-in?

In [Plug-in coding conventions](https://www.dita-ot.org/dev/topics/plugin-coding-conventions.html), we suggest a variant of № 1 above:

> Store the source code of your plug-ins outside of the DITA-OT installation directory, and add the repository location to the list of plug-in directories defined in the `plugindirs` entry of the `configuration.properties` file.

We [suggest this approach](https://www.dita-ot.org/dev/parameters/configuration-properties-file.html#configuration__plugindirs) for testing the sample plug-ins provided with the docs.

I use this configuration properties approach, and symbolic links.

File: resources/source-files.ditamap [hunk 4]
```diff
@@ -138,6 +138,7 @@
   <keydef keys="plugin-anttarget" href="../topics/plugin-anttarget.dita"/>
   <keydef keys="plugin-benefits" href="../topics/plugin-benefits.dita"/>
   <keydef keys="plugin-coding-conventions" href="../topics/plugin-coding-conventions.dita"/>
+  <keydef keys="plugin-common" href="../topics/plugin-common.dita"/>
```
Comment (infotexture) [inline resources/source-files.ditamap:141]: Not sure we need a new file for this.

We have an existing warehouse topic for re-usable task components: [conref-task.dita](https://github.com/dita-ot/docs/blob/develop/resources/conref-task.dita).

File: topics/plugin-common.dita (hunk 3 above)
Comment (infotexture) [reply topics/plugin-common.dita:9]: Right. Might make sense to standardize this and re-use the instructions as you suggest here. 

So far, the other topics in this section all just use verbatim copies of this sentence.

I'd be inclined to recommend the `configuration.properties` approach that we suggest earlier.

File: resources/source-files.ditamap (hunk 4 above)
Comment (jelovirt) [reply resources/source-files.ditamap]: Changed to use conref-task.dita instead.

**Review by infotexture — APPROVED** [review]

**Pull request merged**